webpack/assets/javascripts/react_app/components/Loading/Loading.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Bullseye,
+  Title,
+  EmptyState,
+  EmptyStateIcon,
+  Spinner,
+} from '@patternfly/react-core';
+
+// Centered patternfly 4 loading icon
+const Loading = ({ textSize, iconSize, showText }) => (
+  <Bullseye>
+    <EmptyState>
+      <EmptyStateIcon
+        variant="container"
+        component={() => <Spinner size={iconSize} aria-label="loading icon" />}
```

Comment:
why wrap it in a function?

Reply:
Theoretically it should work, but I get this error without the function:

```
Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of `EmptyStateIcon`.
    at http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:450388:34
    at createFiberFromTypeAndProps (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:450391:13)
    at createFiberFromElement (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:450412:15)
    at reconcileSingleElement (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:440882:23)
    at reconcileChildFibers (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:440939:35)
    at reconcileChildren (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:443043:28)
    at updateHostComponent (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:443522:3)
    at beginWork$1 (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:444862:14)
    at HTMLUnknownElement.callCallback (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:426699:14)
    at Object.invokeGuardedCallbackDev (http://192.168.121.72:6006/foreman-vendor.bundle-v4.11.1-development-8f1002a20f1552db88e3.js:426749:16)
```

Reply:
sounds like Patternfly is doing something weird in that component's code.. Looks like a PF bug, also just another pitfall of having JSX props.  If we had time I'd say look into it and bug them, but you know..

Reply:
yeah it's a patternfly 4 issue (the `<AnyComponent />` renders from function, not from a jsx component)
anyway, how about moving the this anonymous function  outside the return scope?
```js
const Loading = ({ textSize, iconSize, showText }) => {
  const LoadingSpinner = () => <Spinner size={iconSize} aria-label="loading icon" />;
  return (
    <Bullseye>
      <EmptyState>
        <EmptyStateIcon variant="container" component={LoadingSpinner} />
        {showText && (
          <Title size={textSize} headingLevel="h4">
            {__('Loading')}
          </Title>
        )}
      </EmptyState>
    </Bullseye>
  );
};
```

Reply:
@amirfefer thanks, updated!